src/containers/course-section/CourseSectionContainer.tsx

```diff
@@ -90,24 +95,42 @@ const CourseSectionContainer: FC<SectionProps> = ({
         ...sectionData.quizzes,
         ...sectionData.attachments
       ]
-      const updatedResourcesItems = prevResources.filter((prevResource) =>
-        allResourcesItems.some(
-          (currentResource) => currentResource._id === prevResource._id
+
+      const displayOrder = sectionData.order
+
+      const updatedResourcesItems = prevResources
+        .filter((prevResource) =>
+          allResourcesItems.some(
+            (currentResource) => currentResource._id === prevResource._id
+          )
         )
-      )
+        .map((prevResource) => ({
+          ...prevResource,
+          order: displayOrder.indexOf(prevResource._id)
+        }))
+        .sort((a, b) => a.order - b.order)
+
       for (const currentResource of allResourcesItems) {
         if (
           !updatedResourcesItems.some(
             (prevResource) => prevResource._id === currentResource._id
           )
         ) {
-          updatedResourcesItems.push(currentResource)
+          updatedResourcesItems.push({
+            ...currentResource,
+            order: displayOrder.indexOf(currentResource._id)
+          })
         }
       }
       return updatedResourcesItems
     })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I don't recommend skipping any of eslint rules.

Reply:
After refactoring of setResources there was no need for this line, so removed